### PR TITLE
[Core/HLE/ELF/MIPS/GPU/Windows] Fixed zeroise memset sizes

### PR DESCRIPTION
**Pull request by GermanAizek**
@hrydgard,
I went through all memset with zero and yes, mistakes were often made, memset itself is very tricky and easy to make mistakes, but zeroing may not greatly affect the result, my behavior has not changed when testing games.

More info: https://stackoverflow.com/questions/3389464/initializing-a-structure-array-using-memset

### CODE REVIEW

File: Core/MIPS/IR/IRRegCache.cpp [hunk 1]
```diff
@@ -48,8 +48,8 @@ void IRImmRegCache::Discard(IRReg rd) {
 }
 
 IRImmRegCache::IRImmRegCache(IRWriter *ir) : ir_(ir) {
-	memset(&isImm_, 0, sizeof(isImm_));
-	memset(&immVal_, 0, sizeof(immVal_));
+	memset(isImm_, 0, sizeof(isImm_));
+	memset(immVal_, 0, sizeof(immVal_));
```
Comment (hrydgard) [inline Core/MIPS/IR/IRRegCache.cpp:52]: Here's you're doing a change in the right direction, while the one in RegCache.cpp does the opposite. Confused.

File: GPU/GPUCommonHW.cpp [hunk 2]
```diff
@@ -360,7 +360,7 @@ const CommonCommandTableEntry commonCommandTable[] = {
 };
 
 GPUCommonHW::GPUCommonHW(GraphicsContext *gfxCtx, Draw::DrawContext *draw) : GPUCommon(gfxCtx, draw) {
-	memset(cmdInfo_, 0, sizeof(cmdInfo_));
+	memset(&cmdInfo_, 0, sizeof(cmdInfo_));
```
Comment (hrydgard) [inline GPU/GPUCommonHW.cpp:363]: same here

File: GPU/Software/SoftGpu.cpp [hunk 3]
```diff
@@ -399,7 +399,7 @@ SoftGPU::SoftGPU(GraphicsContext *gfxCtx, Draw::DrawContext *draw)
 	fb.data = Memory::GetPointerWrite(0x44000000); // TODO: correct default address?
 	depthbuf.data = Memory::GetPointerWrite(0x44000000); // TODO: correct default address?
 
-	memset(softgpuCmdInfo, 0, sizeof(softgpuCmdInfo));
+	memset(&softgpuCmdInfo, 0, sizeof(softgpuCmdInfo));
```
Comment (hrydgard) [inline GPU/Software/SoftGpu.cpp:402]: and here

File: UI/DevScreens.h [hunk 4]
```diff
@@ -120,7 +120,7 @@ class SystemInfoScreen : public TabbedUIDialogScreenWithGameBackground {
 class AddressPromptScreen : public PopupScreen {
 public:
 	AddressPromptScreen(std::string_view title) : PopupScreen(title, "OK", "Cancel"), addrView_(NULL), addr_(0) {
-		memset(buttons_, 0, sizeof(buttons_));
+		memset(buttons_, 0, sizeof(*buttons_));
```
Comment (hrydgard) [inline UI/DevScreens.h:123]: this is also wrong, revert it.

**Review by hrydgard — CHANGES_REQUESTED** [review]

File: UI/DevScreens.h (hunk 4 above)
Comment (GermanAizek) [reply UI/DevScreens.h]: fix it https://github.com/hrydgard/ppsspp/pull/18971/commits/39af23aefa8517e17b9ade1d4c065c47d50901e7

File: GPU/Software/SoftGpu.cpp (hunk 3 above)
Comment (GermanAizek) [reply GPU/Software/SoftGpu.cpp]: fix it https://github.com/hrydgard/ppsspp/pull/18971/commits/39af23aefa8517e17b9ade1d4c065c47d50901e7

File: GPU/GPUCommonHW.cpp (hunk 2 above)
Comment (GermanAizek) [reply GPU/GPUCommonHW.cpp]: fix it https://github.com/hrydgard/ppsspp/pull/18971/commits/39af23aefa8517e17b9ade1d4c065c47d50901e7

File: Core/MIPS/IR/IRRegCache.cpp (hunk 1 above)
Comment (GermanAizek) [reply Core/MIPS/IR/IRRegCache.cpp]: fix it https://github.com/hrydgard/ppsspp/pull/18971/commits/39af23aefa8517e17b9ade1d4c065c47d50901e7

File: Core/HLE/sceKernelThread.cpp [hunk 5]
```diff
@@ -920,7 +920,7 @@ void __KernelThreadingInit()
 	u32 blockSize = sizeof(idleThreadCode) + ARRAY_SIZE(threadHacks) * 2 * 4;  // The thread code above plus 8 bytes per "hack"
 
 	dispatchEnabled = true;
-	memset(waitTypeFuncs, 0, sizeof(waitTypeFuncs));
+	memset(&waitTypeFuncs, 0, sizeof(waitTypeFuncs));
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply Core/HLE/sceKernelThread.cpp]: fix it https://github.com/hrydgard/ppsspp/pull/18971/commits/39af23aefa8517e17b9ade1d4c065c47d50901e7

File: Core/HLE/sceAtrac.cpp [hunk 6]
```diff
@@ -665,7 +665,7 @@ static u32 atracLibCrc = 0;
 
 void __AtracInit() {
 	atracInited = true;
-	memset(atracIDs, 0, sizeof(atracIDs));
+	memset(atracIDs, 0, sizeof(*atracIDs));
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply Core/HLE/sceAtrac.cpp]: fix it https://github.com/hrydgard/ppsspp/pull/18971/commits/39af23aefa8517e17b9ade1d4c065c47d50901e7

File: Core/HLE/ThreadQueueList.h [hunk 7]
```diff
@@ -163,7 +163,7 @@ struct ThreadQueueList {
 			if (queues[i].data != nullptr)
 				free(queues[i].data);
 		}
-		memset(queues, 0, sizeof(queues));
+		memset(&queues, 0, sizeof(queues));
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply Core/HLE/ThreadQueueList.h]: fix it https://github.com/hrydgard/ppsspp/pull/18971/commits/39af23aefa8517e17b9ade1d4c065c47d50901e7

**Comment by hrydgard** [comment]
decided it's not worth it, closing.

**Pull request closed without merging**